Description: fix for js-yaml ≥ 4
Author: Yadd <yadd@debian.org>
Forwarded: no
Last-Update: 2021-11-24

--- a/lib/command/help.js
+++ b/lib/command/help.js
@@ -32,7 +32,7 @@
         formatPara('You only need to specify the keys that you want to override. Your overrides will be merged ' +
             'with the default config.'));
     console.error('\nThe default configuration is as follows:\n');
-    console.error(yaml.safeDump(configuration.defaultConfig(), { indent: 4, flowLevel: 3 }));
+    console.error(yaml.dump(configuration.defaultConfig(), { indent: 4, flowLevel: 3 }));
     console.error('\n' +
         formatPara('The `watermarks` section does not have a command line equivalent. It allows you to set up ' +
             'low and high watermark percentages for reporting. These are honored by all reporters that colorize ' +
--- a/lib/config.js
+++ b/lib/config.js
@@ -381,7 +381,7 @@
     if (config.verbose) {
         console.error('Using configuration');
         console.error('-------------------');
-        console.error(yaml.safeDump(config, { indent: 4, flowLevel: 3 }));
+        console.error(yaml.dump(config, { indent: 4, flowLevel: 3 }));
         console.error('-------------------\n');
     }
     this.verbose = config.verbose;
@@ -432,7 +432,7 @@
             console.error('Loading config: ' + file);
         }
         configObject = file.match(YML_PATTERN) ?
-            yaml.safeLoad(fs.readFileSync(file, 'utf8'), { filename: file }) :
+            yaml.load(fs.readFileSync(file, 'utf8'), { filename: file }) :
             require(path.resolve(file));
     }
 
