Description: add fix for signal-exit >= 4
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2023-08-17

--- a/nyc/index.js
+++ b/nyc/index.js
@@ -14,7 +14,7 @@
 const { ProcessInfo, ProcessDB } = require('istanbul-lib-processinfo')
 const mkdirp = require('make-dir')
 const Module = require('module')
-const onExit = require('signal-exit')
+const { onExit } = require('signal-exit')
 const path = require('path')
 const rimraf = promisify(require('rimraf'))
 const SourceMaps = require('./lib/source-maps')
--- a/nyc/self-coverage-helper.js
+++ b/nyc/self-coverage-helper.js
@@ -5,7 +5,7 @@
 const path = require('path')
 const fs = require('fs')
 const mkdirp = require('make-dir')
-const onExit = require('signal-exit')
+const { onExit } = require('signal-exit')
 
 module.exports = {
   registered: false,
--- a/spawn-wrap/index.js
+++ b/spawn-wrap/index.js
@@ -13,7 +13,7 @@
 const makeDir = require('make-dir')
 const rimraf = require('rimraf')
 const path = require('path')
-const signalExit = require('signal-exit')
+const { onExit } = require('signal-exit')
 const {IS_DEBUG, debug} = require("./lib/debug")
 const munge = require("./lib/munge")
 const homedir = require("./lib/homedir")
@@ -125,7 +125,7 @@
   }, null, 2) + '\n'
 
   if (!IS_DEBUG) {
-    signalExit(() => rimraf.sync(workingDir))
+    onExit(() => rimraf.sync(workingDir))
   }
 
   makeDir.sync(workingDir)
--- a/spawn-wrap/test/basic.js
+++ b/spawn-wrap/test/basic.js
@@ -3,7 +3,7 @@
 var winNoShebang = IS_WINDOWS && 'no shebang execution on windows'
 var winNoSig = IS_WINDOWS && 'no signals get through cmd'
 
-var onExit = require('signal-exit')
+var { onExit } = require('signal-exit')
 var cp = require('child_process')
 var fixture = require.resolve('./fixtures/script.js')
 var npmFixture = require.resolve('./fixtures/npm')
