Description: fix for recent uuid
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-10-06

--- a/istanbul-lib-processinfo/index.js
+++ b/istanbul-lib-processinfo/index.js
@@ -1,5 +1,5 @@
 'use strict'
-const uuid = require('uuid/v4')
+const { v4: uuid } = require('uuid')
 const archy = require('archy')
 const libCoverage = require('istanbul-lib-coverage')
 const {dirname, resolve} = require('path')
